packages/suite/src/components/suite/modals/ReduxModal/DeviceContextModal/SignMessageModal.tsx

```diff
@@ -0,0 +1,107 @@
+import { selectConnectPopupCall } from '@suite-common/connect-popup';
+import { TrezorDevice } from '@suite-common/suite-types';
+import { selectDeviceAccounts } from '@suite-common/wallet-core';
+import { Card, Column, DotIndicator, H4, NewModal, Row, Text } from '@trezor/components';
+import TrezorConnect from '@trezor/connect';
+import { CoinLogo, ConfirmOnDevice } from '@trezor/product-components';
+import { spacings } from '@trezor/theme';
+
+import { AccountLabel } from 'src/components/suite/AccountLabel';
+import { ConnectCallSource } from 'src/components/suite/ConnectCallSource';
+import { Translation } from 'src/components/suite/Translation';
+import { useSelector } from 'src/hooks/suite';
+import { selectAccountLabels } from 'src/reducers/suite/metadataReducer';
+
+import { ConfirmActionModal } from './ConfirmActionModal';
+
+interface SignMessageModalProps {
+    device: TrezorDevice;
+    message: string;
+    coin?: string;
+    serializedPath?: string;
+}
+
+export const SignMesageModal = ({
+    device,
+    message,
+    coin,
+    serializedPath,
+}: SignMessageModalProps) => {
+    const popupCall = useSelector(selectConnectPopupCall);
+    const accounts = useSelector(selectDeviceAccounts);
+    const accountLabels = useSelector(selectAccountLabels);
+    const deviceModelInternal = device.features?.internal_model;
+
+    if (!popupCall || popupCall?.state !== 'ongoing') return <ConfirmActionModal device={device} />;
+
+    const onCancel = () => {
+        TrezorConnect.cancel();
+    };
+
+    // Connect's coin shortcut (uppercase) <-> Suite's network symbol (lowercase)
+    const networkSymbol = coin?.toLowerCase();
+    const account = accounts.find(a => a.symbol === networkSymbol && a.path === serializedPath);
+
+    return (
+        <NewModal.Backdrop>
+            <ConfirmOnDevice
+                title={<Translation id="TR_CONFIRM_ON_TREZOR" />}
+                deviceModelInternal={deviceModelInternal}
+                deviceUnitColor={device?.features?.unit_color}
+                successText={<Translation id="TR_CONFIRMED_TX" />}
+                onCancel={onCancel}
+            />
+            <NewModal.ModalBase
+                size="small"
+                heading={
+                    popupCall.method === 'ethereumSignTypedData' ? (
+                        <Translation id="TR_SIGN_EIP712_TYPED_DATA" />
+                    ) : (
+                        <Translation id="TR_SIGN_MESSAGE" />
+                    )
+                }
+                description={
+                    <Row
+                        columnGap={spacings.md}
+                        rowGap={spacings.xxs}
+                        flexWrap="wrap"
+                        margin={{ top: spacings.xs }}
+                    >
+                        {account && (
+                            <Row gap={spacings.xxs}>
+                                <CoinLogo size={14} symbol={account.symbol} />
```

Comment:
another nipick
If account is undefined, then no coin symbol. Could that happen? I can imagine that we could maybe show coin symbol even in cases where we don't know the account.

Reply:
Updated, will show coin symbol + name (if supported by Suite)